scripts/apps/templates/controllers/CreateTemplateController.ts

```diff
@@ -32,8 +35,25 @@ export function CreateTemplateController(
     this.types = templates.types;
     this.createTypes = _.filter(templates.types, (element) => element._id !== 'kill');
     this.save = save;
+    this.dateTimeFields = null;
 
     activate();
+    function itemData() {
+        const _item = JSON.parse(JSON.stringify(templates.pickItemData(item)));
+
+        self.dateTimeFields?.forEach((field) => {
```

Comment:
`onTemplateCreate` must be triggered for all extensions not only date-times.

Reply:
Did you mean that we should create the `onTemplateCreate?(field: any): any` interface inside the `IExtensionActivationResult` interface?

Reply:
No. The problem is that you call `onTemplateCreate` inside `forEach` loop. If there are no date time fields, it will never be called, even though it should.

Reply:
As I can see our main goal is that `onTemplateCreate `should be called from all the extensions independently. so what if we define on`TemplateCreate?(field: any): any` in `IExtensionActivationResult` interface then we can use this function by defining in any extension. Is there any approach where we can reflect this function in every extension?

Reply:
Our main goal is to keep implementation details of date-time field inside the extension. Extensions won't call `onTemplateCreate` directly, but will only provide a function, which will be called by core before creating the template.

Keeping `onTemplateCreate` in `ICustomFieldType` is the best for this use case in my opinion, because it's directly related to custom field. If one extension had 5 custom fields, it could provide 5 `onTemplateCreate` functions conveniently. If we needed a global `onTemplateCreate` that is independent from custom fields, we could as well add it to `IExtensionActivationResult['entities']`, but since we are only dealing with a custom field now, I don't think we need it yet.

Reply:
To solve the issue I commented on in the first comment in this thread, you need to remove iteration on datetime fields(`self.dateTimeFields?.forEach`) and instead iterate extensions. For each extension you should get it's config and pass as an argument when calling `onTemplateCreate`